client/lib/domains/registration/availability-messages.js

```diff
@@ -13,14 +12,123 @@ import { translate } from 'i18n-calypso';
 import { getTld } from 'lib/domains';
 import support from 'lib/url/support';
 import { domainAvailability } from 'lib/domains/constants';
+import paths from 'my-sites/domains/paths';
 
-function getAvailabilityNotice( domain, error ) {
+function getAvailabilityNotice( domain, error, site ) {
 	let message,
 		severity = 'error';
 
 	const tld = getTld( domain );
 
 	switch ( error ) {
+		case domainAvailability.REGISTERED:
+			message = translate(
+				"{{strong}}%(domain)s{{/strong}} is already connected to someone else's WordPress.com site.",
+				{
+					args: { domain },
+					components: {
+						strong: <strong />,
+					},
+				}
+			);
+			break;
+		case domainAvailability.REGISTERED_SAME_SITE:
+			message = translate( '{{strong}}%(domain)s{{/strong}} is already registered on this site.', {
+				args: { domain },
+				components: {
+					strong: <strong />,
+				},
+			} );
+			break;
+		case domainAvailability.REGISTERED_OTHER_SITE_SAME_USER:
+			message = translate(
+				'{{strong}}%(domain)s{{/strong}} is already registered on your site %(site)s. Do you want to move it to this site? ' +
+					'{{a}}Yes, move it to this site.{{/a}}',
+				{
+					args: { domain, site },
+					components: {
+						strong: <strong />,
+						a: (
+							<a
+								rel="noopener noreferrer"
+								href={ paths.domainManagementTransferToOtherSite( site, domain ) }
+							/>
+						),
+					},
+				}
+			);
+			break;
+		case domainAvailability.MAPPED_SAME_SITE_TRANSFERRABLE:
+			message = translate(
+				'{{strong}}%(domain)s{{/strong}} is already connected to this site, but registered somewhere else. Do you want to move ' +
+					'it from your current domain provider to WordPress.com so you can manage the domain and the site ' +
+					'together? {{a}}Yes, transfer it to WordPress.com.{{/a}}',
+				{
+					args: { domain },
+					components: {
+						strong: <strong />,
+						a: <a rel="noopener noreferrer" href={ paths.domainTransferIn( site, domain ) } />,
+					},
+				}
+			);
+			break;
+		case domainAvailability.MAPPED_SAME_SITE_NOT_TRANSFERRABLE:
+			message = translate( '{{strong}}%(domain)s{{/strong}} is already connected to this site.', {
+				args: { domain },
+				components: {
+					strong: <strong />,
+				},
+			} );
+			break;
+		case domainAvailability.MAPPED_OTHER_SITE_SAME_USER:
+			message = translate(
+				'{{strong}}%(domain)s{{/strong}} is already connected to your site %(site)s. If you want to connect it to this site ' +
```

Comment:
If you trigger this notice during signup, you'll get this notice:
<img width="992" alt="screen shot 2017-12-12 at 16 03 00" src="https://user-images.githubusercontent.com/3392497/33891395-1a1df05e-df56-11e7-951a-9cb39c5d9c29.png">

Notice the missing site and, again, a mention of "this site".